Frontend/src/pages/Signin.js

```diff
@@ -0,0 +1,326 @@
+import React, { useEffect, useState } from 'react'
+import { batch, useDispatch, useSelector } from 'react-redux'
+import { useNavigate } from 'react-router-dom'
+import styled from 'styled-components'
+
+import Avatar from '../components/Avatar'
+
+import user from 'reducers/user'
+
+import { API_URL } from '../utilis/urls'
+
+const Signin = () => {
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [email, setEmail] = useState('')
+  const [mode, setMode] = useState('signin')
+  const [avatar, setAvatar] = useState('tourism')
+
+  const accessToken = useSelector((store) => store.user.accessToken)
+
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+
+  const avatars = ['tourism', 'bear', 'man', 'woman', 'user']
+
+  useEffect(() => {
+    if (accessToken) {
+      navigate('/user')
+    }
+  }, [accessToken, navigate])
+
+  const onHandleSignIn = (event) => {
+    event.preventDefault()
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        username,
+        password,
+        email,
+        avatar,
+      }),
+    }
+
+    fetch(API_URL(`users/${mode}`), options)
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.success) {
+          batch(() => {
+            dispatch(user.actions.setUserId(data.response.userId))
+            dispatch(user.actions.setUsername(data.response.username))
+            dispatch(user.actions.setAccessToken(data.response.accessToken))
+            dispatch(user.actions.setAvatar(data.response.avatar))
+            dispatch(user.actions.setEmail(data.response.email))
+            dispatch(user.actions.setError(null))
+          })
+        } else {
+          batch(() => {
+            dispatch(user.actions.setUserId(null))
+            dispatch(user.actions.setUsername(null))
+            dispatch(user.actions.setAccessToken(null))
+            if (mode === 'signin') {
+              dispatch(user.actions.setError(data.response))
+            } else if (mode === 'signup') {
+              dispatch(user.actions.setError(data.response.code))
+            } else {
+              console.log('Problem')
```

Comment:
Once again, `console.log` should be replaced with some meaningful message in layout